bentoml/_internal/frameworks/tensorflow_v1.py

```diff
@@ -611,72 +527,50 @@ def required_models(self) -> t.List[Tag]:
 
     @property
     def _num_threads(self) -> int:
-        if _is_gpu_available() and self.resource_quota.on_gpu:
+        if is_gpu_available() and self.resource_quota.on_gpu:
             return 1
         return int(round(self.resource_quota.cpu))
 
     @property
     def num_replica(self) -> int:
-        if _is_gpu_available() and self.resource_quota.on_gpu:
+        if is_gpu_available() and self.resource_quota.on_gpu:
             return len(tf.config.list_physical_devices("GPU"))
         return 1
 
     # pylint: disable=attribute-defined-outside-init
     def _setup(self) -> None:
         # setup a global session for model runner
-        self._session = tf.compat.v1.Session(
-            config=tf.compat.v1.ConfigProto(**self._config_proto)
-        )
         self._model = load(self._tag, model_store=self._model_store)
-        if not TF2:
-            raw_predict_fn = self._model.signatures["serving_default"]
-        else:
-            raw_predict_fn = getattr(self._model, self._predict_fn_name)
+        raw_predict_fn: t.Callable[..., t.Any] = self._model.signatures[
+            "serving_default"
+        ]
         self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
 
     def _run_batch(
-        self,
-        *args: t.Union[
-            t.List[t.Union[int, float]], "np.ndarray[t.Any, np.dtype[t.Any]]", tf.Tensor
-        ],
-        **kwargs: t.Union[
-            t.List[t.Union[int, float]], "np.ndarray[t.Any, np.dtype[t.Any]]", tf.Tensor
-        ],
-    ) -> "np.ndarray[t.Any, np.dtype[t.Any]]":
-        params = Params[
-            t.Union[
-                t.List[t.Union[int, float]],
-                "np.ndarray[t.Any, np.dtype[t.Any]]",
-                tf.Tensor,
-            ]
-        ](*args, **kwargs)
+        self, *args: "TFArgType", **kwargs: "TFArgType"
+    ) -> "tf_ext.TensorLike":
+        params = Params["TFArgType"](*args, **kwargs)
 
         with tf.device(self._device_id):
 
-            def _mapping(
-                item: t.Union[
-                    t.List[t.Union[int, float]],
-                    "np.ndarray[t.Any, np.dtype[t.Any]]",
-                    tf.Tensor,
-                ]
-            ) -> tf.Tensor:
-                if not isinstance(item, tf.Tensor):
-                    return tf.convert_to_tensor(item, dtype=tf.float32)
+            def _mapping(item: "TFArgType") -> "tf_ext.TensorLike":
+                if not LazyType["tf_ext.TensorLike"]("tf.Tensor").isinstance(item):
+                    return t.cast(
+                        "tf_ext.TensorLike",
+                        tf.convert_to_tensor(item, dtype=tf.float32),
```

Comment:
why float32 here?

Reply:
because we don't want to do float64

Reply:
oh i see

Reply:
so should we just not specify the dtype then?